docs/cookbook/rag.md

```diff
@@ -0,0 +1,257 @@
+# Retrieval Augmented Generation (RAG)
+
+When you need to provide additional information to your model that it hasn't been pretrained on, we retrieve the relevant information from an external source and provide it as context to our model.
+
+## A Barebones Retrieval Example
+
+!!! note
+
+    Take a look at the [code](https://github.com/Mirascope/mirascope/blob/main/cookbook/basic_rag_example/local_embeddings.py) in our repo.
+
+Here are four simple statements about the 2022/3 soccer season that our chat model does not know:
+
+```python
+soccer_info = [
+    "Manchester City won the league, continuing their domestic dominance.",
+    "In La Liga, FC Barcelona emerged victorious with a sturdy defense.",
+    "Dortmund slipped up in the last matchday to hand Bayern the Bundesliga title.",
+    "For the first time since Maradona's days, Napoli lifted the Serie A trophy.",
+]
+```
+
+The end goal is to be able to ask our model: which team in {insert_country} won the league in the 2023 season? Then, we want to put this additional context in our prompt to obtain the correct answer.
```

Comment:
the {insert_country} is a bit awkward. maybe a `insert_country`

Reply:
how just `a given country`?